pkgs/by-name/tu/turn-rs/package.nix

```diff
@@ -0,0 +1,36 @@
+{
+  rustPlatform,
+  lib,
+  fetchFromGitHub,
+}:
+
+let
+  pname = "turn-rs";
+  version = "3.0.0";
+  src = fetchFromGitHub{
+    owner = "mycrl";
+    repo = "turn-rs";
+    rev = "refs/tags/v${version}";
+    hash = "sha256-AtNvvL/zNqqfCMVG2o8zrT+a1O6BkvKFf1nsvUDrvWs=";
+  };
+in
+rustPlatform.buildRustPackage {
+  inherit pname version src;
+
+  cargoLock = {
+    lockFile = ./Cargo.lock;
```

Comment:
given turn-rs is an application, can you ask upstream to commit a lockfile ?

some links re lock https://doc.rust-lang.org/nightly/cargo/faq.html#why-have-cargolock-in-version-control

here tracking issue for Cargo.lock https://github.com/NixOS/nixpkgs/issues/327063